include/exadg/fluid_structure_interaction/precice/driver_solid.h

```diff
@@ -0,0 +1,270 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_FLUID_STRUCTURE_INTERACTION_PRECICE_DRIVER_SOLID_H_
+#define INCLUDE_EXADG_FLUID_STRUCTURE_INTERACTION_PRECICE_DRIVER_SOLID_H_
+
+// application
+#include <exadg/fluid_structure_interaction/user_interface/application_base.h>
+
+// utilities
+#include <exadg/functions_and_boundary_conditions/verify_boundary_conditions.h>
+#include <exadg/matrix_free/matrix_free_data.h>
+#include <exadg/utilities/print_general_infos.h>
+
+// grid
+#include <exadg/grid/grid_motion_elasticity.h>
+#include <exadg/grid/grid_motion_poisson.h>
+#include <exadg/poisson/spatial_discretization/operator.h>
+
+// Structure
+#include <exadg/structure/spatial_discretization/operator.h>
+#include <exadg/structure/time_integration/time_int_gen_alpha.h>
+
+namespace ExaDG
+{
+namespace FSI
+{
+namespace preCICE
+{
+template<int dim, typename Number>
+class DriverSolid : public Driver<dim, Number>
+{
+private:
+  using VectorType = typename dealii::LinearAlgebra::distributed::Vector<Number>;
+
+public:
+  DriverSolid(std::string const &                           input_file,
+              MPI_Comm const &                              comm,
+              std::shared_ptr<ApplicationBase<dim, Number>> app,
+              bool const                                    is_test)
+    : Driver<dim, Number>(input_file, comm, app, is_test)
+  {
+    structure = std::make_shared<SolverStructure<dim, Number>>();
+  }
+
+
+  void
+  setup_application()
+  {
+    dealii::Timer timer_local;
+    timer_local.restart();
+
+    this->application->structure->setup();
+
+    this->timer_tree.insert({"FSI", "Setup", "Application"}, timer_local.wall_time());
+  }
+
+
+  void
+  setup_structure()
+  {
+    dealii::Timer timer_local;
+    timer_local.restart();
+
+    structure->setup(this->application->structure, this->mpi_comm, this->is_test);
+
+    this->timer_tree.insert({"FSI", "Setup", "Structure"}, timer_local.wall_time());
+  }
+
+
+
+  void
+  setup_interface_coupling()
+  {
+    this->precice =
+      std::make_shared<ExaDG::preCICE::Adapter<dim, dim, VectorType>>(this->precice_parameters,
+                                                                      this->mpi_comm);
+    // TODO generalize interface handling for multiple interface IDs
+    this->precice->add_write_surface(
+      this->application->structure->get_boundary_descriptor()->neumann_cached_bc.begin()->first,
+      this->precice_parameters.write_mesh_name,
+      {this->precice_parameters.displacement_data_name,
+       this->precice_parameters.velocity_data_name},
+      this->precice_parameters.write_data_type,
+      structure->matrix_free,
+      structure->pde_operator->get_dof_index(),
+      dealii::numbers::invalid_unsigned_int);
+
```

Comment:
In analogy to the fluid driver, I would help the reader in identifying which coupling is set up here, e.g. comments like "// structure to fluid"